torch_geometric/transforms/add_positional_encoding.py

```diff
@@ -0,0 +1,119 @@
+from typing import Optional
+
+import numpy as np
+import torch
+from scipy.sparse.linalg import eigs, eigsh
+from torch_scatter import scatter_add
+from torch_sparse import spspmm
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+from torch_geometric.utils import (
+    get_laplacian,
+    remove_self_loops,
+    to_scipy_sparse_matrix,
+)
+from torch_geometric.utils.num_nodes import maybe_num_nodes
+
+
+@functional_transform('add_positional_encoding')
+class AddPositionalEncoding(BaseTransform):
+    r"""Adds positional encoding to the given graph (functional name:
+    :obj:`add_positional_encoding`).
+
+    Args:
+        name: (str): The type of the positional encoding
+            (:obj:`"laplacian_eigenvector_pe"`, :obj:`"random_walk_pe"`).
+            If set to :obj:`"laplacian_eigenvector_pe"`, will add the
+            Laplacian eigenvector positional encoding from the `"Benchmarking
+            Graph Neural Networks" <https://arxiv.org/abs/2003.00982>`_ paper.
+            If set to :obj:`"random_walk_pe"`, will add the random walk
+            positional encoding from the `"Graph Neural Networks with Learnable
+            Structural and Positional Representations"
+            <https://arxiv.org/abs/2110.07875>`_ paper.
+        num_channels: (int): The number of channels. If :attr:`name` is
+            :obj:`"laplacian_eigenvector_pe"`, it will be the number of the
+            smallest non-trivial eigenvectors. If :attr:`name` is
+            :obj:`"random_walk_pe"`, it will be the number of the random walk
+            steps.
+        method: (str, optional): The method to add positional encoding.
+            If set to :obj:`"attr"`, will be added to the data object
+                as a new attribute.
```

Comment:
I guess this is the doc building error ( also on line 44)

Reply:
Thank you for your help. I just fixed it.